### ISSUE TITLE
Documentation: how do I configure the ISP-delegated prefix?

### ISSUE THREAD

**Issue by Abhayakara** [sbyx/hnetd#29]
Is this the -p argument to hnet-ifup?   What are the meanings of the different interface types?   Could use some examples.

**Comment by sbyx** [comment]
I guess we should expand the documentation here. Mainly automated setup (DHCPv6-PD + RA) is easily accessible in the generic variant. I suppose we should add a shortcut to do manual setup as well.

I think the following should do:

hnet-ifup -c external eth1
hnet-call enable_ipv6_uplink "{\"ifname\": \"eth1\", \"prefix\": ["{\"address\": \"$prefix\", \"preferred\": $preferred, \"valid\": $valid}"], \"dns\": [\"2001:db8::1\"]}"

**Comment by Abhayakara** [comment]
What I'm looking for is not a recipe, but an explanation.   If the way the thing works isn't documented, and all you can go by is a collection of scripts which work for someone somewhere, you don't have a replicable solution.

**Comment by sbyx** [comment]
Acknowledged.  Will come up with a solution tomorrow.

**Comment by Abhayakara** [comment]
On Apr 9, 2015, at 3:40 PM, Steven Barth <notifications@github.com> wrote:
> Acknowledged. Will come up with a solution tomorrow.

Thanks!

**Comment by sbyx** [comment]
I made some additions to the branch dncp-pa-inc (to be merged).
https://github.com/sbyx/hnetd/blob/dncp-pa-inc/README

The modes are now explained better and I added a reference to the HNCP draft for further explanations.
Furthermore hnet-ifup now has 3 new parameters -4 indicating a public IPv4 address (and IPv4-uplink) on the interface, -6 indicating a delegated prefix (and IPv6 uplink) and -D indicating a global recursive DNS servers.

hnet-ifup -c static -6 2001:db8::/56 eth1
should now be sufficient to indicate eth1 being an external interface and IPv6 uplink with given delegated prefix.

As for additional explanation, normally a simple hnet-ifup eth1 would start the interface in automatic border discovery mode (or hnet-ifup -c external eth1 in external mode) and cause a DHCP and DHCPv6(+PD) client to be run by hnetd. Upon a successful binding the client would notify hnetd via hnet-call enable_ipv6_uplink of the PD and uplink status.

PS: as you asked on the homenet ML OpenWrt's https://github.com/sbyx/odhcp6c and https://github.com/sbyx/odhcpd are used as default DHCPv6 client and servers here.
